aarch64/src/trap.rs

```diff
@@ -0,0 +1,65 @@
+use crate::registers::EsrEl1;
+use port::println;
+
+#[cfg(not(test))]
+core::arch::global_asm!(include_str!("trap.S"));
+
+pub fn init() {
+    unsafe { init_interrupts() };
+}
+
+extern "C" {
+    fn init_interrupts();
+}
+
+/// Register frame at time interrupt was taken
+#[derive(Copy, Clone, Debug)]
+#[repr(C)]
+pub struct InterruptFrame {
```

Comment:
Just a minor thing, I'd call this `TrapFrame`.

Reply:
Changed